File: src/main/java/com/pocketempire/config/CasusBelliConfigLoader.java
package com.pocketempire.config;

import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.InputStream;
import java.util.Collection;
import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

public class CasusBelliConfigLoader {
    private static final Map<String, CasusBelliConfig> configs;

    static {
        configs = loadConfigs();
    }

    private static Map<String, CasusBelliConfig> loadConfigs() {
        try {
            ObjectMapper mapper = new ObjectMapper();
            InputStream is = CasusBelliConfigLoader.class.getResourceAsStream("/casus_belli.json");
            CasusBelliConfig[] configs = mapper.readValue(is, CasusBelliConfig[].class);

            Map<String, CasusBelliConfig> map = new HashMap<>();
            for (CasusBelliConfig config : configs) {
                map.put(config.getId(), config);
            }
            return map;
        } catch (Exception e) {
            e.printStackTrace();
            return Collections.emptyMap();
        }
    }

    public static CasusBelliConfig getConfig(String id) {
        return configs.get(id);
    }

    public static Collection<CasusBelliConfig> getAll() {
        return configs.values();
    }
}

File: src/main/kotlin/com/pocketempire/config/CasusBelliConfigLoader.kt
package com.pocketempire.config

import com.fasterxml.jackson.databind.ObjectMapper

object CasusBelliConfigLoader {

    private val configs: Map<String, CasusBelliConfig> = loadConfigs()

    private fun loadConfigs(): Map<String, CasusBelliConfig> {
        return try {
            val inputStream = requireNotNull(
                CasusBelliConfigLoader::class.java
                    .getResourceAsStream("/casus_belli.json")
            ) {
                "Could not find /casus_belli.json"
            }

            inputStream.use {
                val casusBelli = ObjectMapper().readValue(
                    it,
                    Array<CasusBelliConfig>::class.java
                )

                casusBelli.associateBy { config -> config.id }
            }
        } catch (e: Exception) {
            e.printStackTrace()
            emptyMap()
        }
    }

    @JvmStatic
    fun getConfig(id: String): CasusBelliConfig? =
        configs[id]

    @JvmStatic
    fun getAll(): Collection<CasusBelliConfig> =
        configs.values
}
